Automattic/jetpack
# General: update our PHP version requirements to PHP 7.0

Fixes #31714

## Proposed changes:

Now that WordPress 6.4 is available, our plugins will be able to drop support for WordPress 6.2 and require WordPress 6.3 (to happen in a follow-up PR, #34127).

WordPress 6.3 introduced a new PHP version requirement: PHP 7.0. We can now match that requirement in all our tools.

### Other information:


## Jetpack product discussion

* #32865

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Check if CI is happy
* Check that the changes to the codesniffer to remove one set of rules makes sense.
